client/src/helpers/ImagesTreeViewHelpers.js

```diff
@@ -16,20 +16,25 @@ export const DoDataPreprocessing = (filteredActivityMaps, currentExperimentName)
   Object.values(filteredActivityMaps).map(obj => {
     // if experiment key exists
     if (obj.experiment !== undefined) {
-      if (obj.experiment === true) {
+      if(obj.experiment === true){
         // experiment is true
-        obj.hierarchy = obj.hierarchy === undefined || obj.hierarchy.includes(currentExperimentName)
-          ? obj.hierarchy
-          : [currentExperimentName, ...obj.hierarchy];
-      } else {
+        if(obj.hierarchy === undefined){ // no hierarchy key
+          obj.hierarchy = [currentExperimentName]
+        }else{
+          // hierarchy key exists
+          obj.hierarchy = obj.hierarchy.includes(currentExperimentName) ? obj.hierarchy : [currentExperimentName,...obj.hierarchy]
+        }
+      }else {
         // experiment is false
-        obj.hierarchy = ['others'];
+        obj.hierarchy = ['others']
       }
     } else {
       // experiment key not exists
-      obj.hierarchy = obj.hierarchy === undefined || obj.hierarchy.includes(currentExperimentName)
-        ? obj.hierarchy
-        : [currentExperimentName, ...obj.hierarchy];
+      if(obj.hierarchy === undefined){
+        obj.hierarchy =[currentExperimentName]
+      } else {
+        obj.hierarchy = obj.hierarchy.includes(currentExperimentName) ? obj.hierarchy : [currentExperimentName,...obj.hierarchy]
+      }
```

Comment:
```
if (obj.experiment === undefined || obj.experiment === true) {
    // experiment is true
    if(obj.hierarchy === undefined){ // no hierarchy key
      obj.hierarchy = [currentExperimentName]
    } else {
      // hierarchy key exists
      obj.hierarchy = obj.hierarchy.includes(currentExperimentName) ? obj.hierarchy : [currentExperimentName,...obj.hierarchy]
    }
} else {
    // experiment is false
    obj.hierarchy = ['others']
}

return obj;
```

is this doing the same? if yes, could we do a clean up please? Thanks

Reply:
@ddelpiano yes this works but I have added a small modification that would cover the case where we have the experiment is false but we have a hierarchy key exists, in the above one will consider it as  ['others'] but I have made the modification to take the original  hierarchy array without any changes